modules/guides/guides.routing.yml

```diff
@@ -7,8 +7,8 @@ guides.list:
     _permission: 'access guides'
 
 guides.guide:
-  path: '/admin/guides/{filename}'
+  path: '/admin/guides/{subdirectory}/{filename}'
```

Comment:
Let's continue [this](https://github.com/Pronovix/devportal-drupal-module/pull/85#discussion_r314626298) discussion here.

> : it doesn't work with {path} and it always throws an InvalidParameterException, because "Route parameters can contain any values except the / slash character, because that's the character used to separate the different parts of the URLs."

It means` {subdirectory}` cannot contain either a slash either, isn't it? IOW `admin/guides/foo/bar/bar` where `foo/bar` is a subdirectory is not valid, is it?

Reply:
Yes, that's true, I tried it out and if I create a new subdirectory with a guide file in a subdirectory, then it doesn't appear in the list of the guides, but as I read in the README file of the module, I found that that there shouldn't be deeper subdirectories in the first level subdirectories, so I thought it is the aim: `2. Create a folder where you use underscores instead of spaces, e.g. `User_guide`, this will be the link appearing on `/admin/guides`
3. Create a markdown file inside the guide folder, e.g. `user-guide.md`, this will be the url for the guide: `admin/guides/user-guide``

Reply:
@gaborkurucz do we want to support subfolders or not? 
Remember we had a discussion about extending this module in the future to support discovering guides from modules: https://github.com/Pronovix/devportal-drupal-module/pull/85#issuecomment-518541878

Reply:
If this exception came from Symfony directly, according to Symfony's doc it should be possible to allow slashes in routes.
https://symfony.com/doc/3.4/routing/slash_in_parameter.html

Reply:
According to this reply: https://github.com/Pronovix/devportal-drupal-module/pull/85#issuecomment-518561302
and if I were to make the decision then it's still Option 2.

However, now that you mention subfolders do you mean navigation depth? :) I think that would be a good idea later on, but I think first it's enough to collect folders from modules. Maybe do it similarly as https://www.drupal.org/docs/8/core/modules/help/help-module-overview?

Have the categories as module names, and when you click on a module you get the list of guides that module defines.

Reply:
> However, now that you mention subfolders do you mean navigation depth?

Yes, it means navigation depth represented in URLs well, which means adding `/` to routes that are not supported OOTB.

I think the best we can do for now is getting rid of the `{subdirectory}/{filename}` path (and method parameters for now and using a simple `{path}` parameter that could cover the current "single level" use case and hopefully it could also support discovering guides from modules later. cc @morvaim2